hash/structhash/structhash_test.go

```diff
@@ -69,6 +69,37 @@ func TestDump(t *testing.T) {
 			}{nil},
 			"{a:nil}",
 		},
+		{
```

Comment:
The following struct returns `{a:{}}` instead of `{}`.
If the content of the interface doesn't return an hash, the interface should not be in the hash at all. What do you think?
```suggestion
		{
			struct{ a interface{} }{
				a: struct{ b int }{
					b: 1,
				},
			},
			"{}",
		},
		{
```

Reply:
I think no, as it would introduce complexity. Also, it would require to keep a recursive stack to make a decision.

Reply:
Done in https://github.com/mesg-foundation/engine/pull/1319 with a simple solution.